x/update_validators/model.go

```diff
@@ -61,10 +62,26 @@ func (m *Accounts) Copy() orm.CloneableData {
 func (m *Accounts) Validate() error {
 	return AsWeaveAccounts(m).Validate()
 }
+
 func GetAccounts(bucket orm.Bucket, kv weave.KVStore) (orm.Object, error) {
 	return bucket.Get(kv, []byte(Key))
 }
 
+func HasPermission(object orm.Object, addr weave.Address) (bool, error) {
+	switch t := object.Value().(type) {
+	case *Accounts:
+		accts := AsWeaveAccounts(t)
+		for _, v := range accts.Addresses {
+			if v.Equals(addr) {
+				return true, nil
+			}
+		}
+	default:
+		return false, ErrWrongType(reflect.TypeOf(t).Name())
```

Comment:
I would push this stringification into the ErrWrongType (which probably will be used multiple places). This is similar to the following standard error: https://github.com/confio/weave/blob/master/errors/common.go#L66-L70

Maybe we should generalize that error not just to refer to transactions, or add you other ErrWrongType to the common errors.

Reply:
Fair point

Reply:
Weave/errors/common.go

That was what I referred to. 
But feel free to keep this local

Reply:
Ah, gotcha. It's a bit confusing, because you could also consider common whatever lives in the root of `x` folder. I think this needs to be re-arranged somehow as it basically has all sorts of things mixed together there.